src/app/detail/ui/ActivityWrapper.tsx

```diff
@@ -10,29 +10,108 @@ import { postComment, deleteComment } from '../[task_id]/[user_type]/actions'
 import { CreateComment } from '@/types/dto/comment.dto'
 import { fetcher } from '@/utils/fetcher'
 import useSWR from 'swr'
-import { useSearchParams } from 'next/navigation'
-import { useEffect, useState } from 'react'
+import { useDeferredValue, useEffect, useState } from 'react'
+import { useSelector } from 'react-redux'
+import { selectTaskBoard } from '@/redux/features/taskBoardSlice'
+import { selectAuthDetails } from '@/redux/features/authDetailsSlice'
+import { generateRandomString } from '@/utils/generateRandomString'
+import { mutate } from 'swr'
+
+export type TempCommentType = {
+  id: string
+  type: ActivityType // Assuming you only handle comments
+  initiator: {
+    id?: string // User details may be incomplete in temp comments
+    givenName?: string
+    familyName?: string
+    avatarImageUrl?: string
+  }
+  details: {
+    content: string
+  }
+  createdAt: string
+}
 
 export const ActivityWrapper = ({ token, task_id }: { token: string; task_id: string }) => {
   const { data: activities, isLoading } = useSWR(`/api/tasks/${task_id}/activity-logs/?token=${token}`, fetcher, {
-    refreshInterval: 1000,
+    refreshInterval: 10_000,
   })
+  const { tokenPayload } = useSelector(selectAuthDetails)
+  const { assignee } = useSelector(selectTaskBoard)
+  const currentUserId = tokenPayload?.clientId ?? tokenPayload?.internalUserId
+  const currentUserDetails = assignee.find((el) => el.id === currentUserId)
+
+  const handleCreateComment = async (postCommentPayload: CreateComment) => {
+    const tempComment: TempCommentType = {
+      type: ActivityType.COMMENT_ADDED,
+      id: generateRandomString('temp-comment'),
+      initiator: {
+        id: currentUserDetails?.id,
+        givenName: currentUserDetails?.givenName,
+        familyName: currentUserDetails?.familyName,
+        avatarImageUrl: currentUserDetails?.avatarImageUrl,
+      },
+      details: {
+        content: postCommentPayload.content,
+      },
+      createdAt: new Date().toISOString(),
+    }
+
+    // Optimistically update the SWR cache
+    const cacheKey = `/api/tasks/${task_id}/activity-logs/?token=${token}`
 
-  const [isFirstPageLoad, setIsFirstPageLoad] = useState(true)
+    // Optimistic UI update
+    mutate(
+      cacheKey,
+      (currentData: { data: (LogResponse | TempCommentType)[] } | undefined) => {
```

Comment:
hmmm... tbh I dnt understand why we need `TempCommentType` and cant use `LogResponse` directly. But Im probably just missing some context. This comment is not blocking review/merge, just adding my 2 cents

Reply:
We can make use of `LogResponse` but need to handle a lot of fields and also need to probably change the current type of `LogResponse`.  I chose to create a new `tempComment` assignment with `TempCommentType` structure to minimize the unnecessary need to handle multiple fields that are not wanted for a short period of optimistic update until a new mutation takes place. @makeitraina

Reply:
is it that complicated? I think the defaults can be set pretty easily right?

Reply:
changed added in the latest commit @makeitraina